app/src/org/commcare/android/tasks/FormRecordCleanupTask.java

```diff
@@ -76,181 +72,156 @@ public FormRecordCleanupTask(Context context, CommCarePlatform platform, int tas
+     * apply any updates found to the form record, such as UUID and date
+     * modified, returning an updated copy.  Write the updated record to
+     * storage. If the record is associated with a case id, recompute and write
+     * the SessionStateDescriptor too.
+     *
+     * @param context   Used to get the filepath of the form instance
+     *                  associated with the record.
+     * @param oldRecord Reparse this record and return an updated copy of it
+     * @param newStatus The new form record status
+     * @param storage   User storage where updated FormRecord is written
+     * @return The reparsed form record and the associated case id, if present
+     * @throws IOException                       Problem opening the saved form
+     *                                           attached to the record.
+     * @throws InvalidStructureException         Occurs during reparsing of the
+     *                                           form attached to record.
+     * @throws XmlPullParserException
+     * @throws UnfullfilledRequirementsException Parsing encountered a platform
+     *                                           versioning problem
+     */
+    public static FormRecord updateAndWriteRecord(Context context,
+                                                  CommCarePlatform platform,
+                                                  FormRecord oldRecord,
+                                                  String newStatus,
+                                                  SqlStorage<FormRecord> storage)
+            throws InvalidStructureException, IOException,
+            XmlPullParserException, UnfullfilledRequirementsException {
+
+        Pair<FormRecord, String> recordUpdates =
+            reparseRecord(context, oldRecord, FormRecord.STATUS_SAVED);
+        FormRecord updated = recordUpdates.first;
+        String caseId = recordUpdates.second;
+
+        if (caseId != null &&
+                FormRecord.STATUS_UNINDEXED.equals(oldRecord.getStatus()) &&
+                !FormRecord.STATUS_UNINDEXED.equals(newStatus)) {
+            // There is a case id associated with an unidexed form record,
+            // calculate the state descripter and write it.
+            //
+            // Should only occur when we are loading forms manually onto the
+            // phone using DataPullTask.
+            AndroidSessionWrapper asw
+                = AndroidSessionWrapper.mockEasiestRoute(platform,
+                        oldRecord.getFormNamespace(), caseId);
+            asw.setFormRecordId(updated.getID());
+
+            SqlStorage<SessionStateDescriptor> ssdStorage =
+                CommCareApplication._().getUserStorage(SessionStateDescriptor.class);
+
+            try {
+                ssdStorage.write(asw.getSessionStateDescriptor());
+            } catch (StorageFullException e) {
+                // TODO PLM should we abort completely here?

```

Comment:
for now we can actually dump the catch, this is now a runtime exception we really don't expect on android.